app/src/main/java/org/listenbrainz/android/application/App.kt

```diff
@@ -24,6 +34,29 @@ class App : Application() {
                 startListenService()
             }
         }
+
+        val token = appPreferences.lbAccessToken
```

Comment:
I think it would be better in dashboard activity. This Application class only gets destroyed when the device switches off or user force stops the app.

Reply:
Yes @07jasjeet Ik that. Most probably we should have the getUser call here. We will replace this api call with the user call then in future. We can time the api call to call itself every 5 secs. That's the idea.

Reply:
Basically we can make the get request every 2mins or so.

Reply:
This is common hack when working with REST apis.

Reply:
Also @07jasjeet , keeping the api call in dashboard will fire it whenever the dashboard is opened which will make things inconsistent.

Reply:
But that does somewhat gives user the belief that they are in control of when the API is called/their internet is used. plus if the app remains dormant for a while, it'll just eat data in background.

Reply:
We could also show a little snackbar that says "details updated" or something to be more transparent.

Reply:
This topic is very two sided tbh. We could use any but the one I mentioned could be an option.
Rest is upto you to decide.

Reply:
The snackbar idea is good. We anyway won't be updating any visible UI changes for the user where they would need a snackbar btw.